src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs

```diff
@@ -814,13 +811,15 @@ public async Task WriteAsync_LocalAbort_Throws()
         public async Task WaitForWriteCompletionAsync_ServerWriteAborted_Throws()
         {
             const int ExpectedErrorCode = 0xfffffff;
+            SemaphoreSlim sem = new SemaphoreSlim(0);
```

Comment:
These 3 lines are not strictly needed for this PR and I'll remove them. But they need to be introduced in stream PR, where we're getting rid of `ShutdownCompleted` (we could replace it with `await Task.WhenAll(ReadsClosed, WritesClosed)` or something similar I guess, I haven't thought about that when fixing the tests), since then you're left with `DisposeAsync` which will attempt to abort read side which will race with `AbortWrite` from the server and sometimes win. Or the semaphore could be replaced with await stream.ReadsClosed on the client side and check for StreamAborted exception.